opentera-webrtc-native-client/OpenteraWebrtcNativeClient/python/package/CMakeLists.txt

```diff
@@ -1,41 +1,118 @@
+#     ../../../../signaling-server/requirements.txt
+#     ../../../../signaling-server/README.md
+# )
+
+# install(FILES ${signaling_server_source_files} DESTINATION python/signaling-server)
+
+set(GENERATE_CONFIGURE_OUTPUT_FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/__init__.py
+    ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/README.md
+    ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/setup.py
+    ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/LICENSE
+)
+
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/so.md5
+    DEPENDS ${PYTHON_LIB_NAME}
+    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:${PYTHON_LIB_NAME}> ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/$<TARGET_FILE_NAME:${PYTHON_LIB_NAME}>
+    COMMAND md5sum ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/$<TARGET_FILE_NAME:${PYTHON_LIB_NAME}> > ${CMAKE_CURRENT_BINARY_DIR}/so.md5
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    VERBATIM
+)
+add_custom_target(
+    python-opentera-webrtc-native-client-so
+    ALL
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/so.md5
+    VERBATIM
+)
 
 # Generate html doc
 if (OPENTERA_WEBRTC_ENABLE_PYTHON_HTML_DOC)
-    add_custom_target(pydoc-html-file ALL
-	    COMMAND python3 -m pydoc -w opentera_webrtc_native_client
-	    WORKING_DIRECTORY ${LIBRARY_OUTPUT_PATH}
-	    DEPENDS opentera_webrtc_native_client)
+    add_custom_command(
+        OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/${PYTHON_LIB_NAME}.html
+        DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/so.md5 ${GENERATE_CONFIGURE_OUTPUT_FILES} python-opentera-webrtc-native-client-so
+        COMMAND /usr/bin/env python3 -m pydoc -w ${PYTHON_LIB_NAME}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+        VERBATIM
+    )
+    add_custom_target(
+        python-opentera-webrtc-native-client-html
+        ALL
+        DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/${PYTHON_LIB_NAME}.html
+        VERBATIM
+    )
+    set(HTML_DOC_DEPENDENCY_OPTION python-opentera-webrtc-native-client-html ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/${PYTHON_LIB_NAME}.html)
+else()
+    set(HTML_DOC_DEPENDENCY_OPTION ${CMAKE_CURRENT_BINARY_DIR}/so.md5 ${GENERATE_CONFIGURE_OUTPUT_FILES} python-opentera-webrtc-native-client-so)
+endif()
 
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    DEPENDS ${HTML_DOC_DEPENDENCY_OPTION}
+    COMMAND /usr/bin/env python3 setup.py bdist bdist_wheel sdist
+    COMMAND tar c ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/dist 2> /dev/null | md5sum > ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    VERBATIM
+)
+add_custom_target(
+    python-opentera-webrtc-native-client-wheel
+    ALL
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    VERBATIM
+)
```

Comment:
Generate wheel AFTER documentation if it is enabled, or AFTER .so file if documentation is disabled